src/wp-includes/theme.php

```diff
@@ -4416,3 +4445,45 @@ static function ( $active, WP_Customize_Panel $panel ) {
 		2
 	);
 }
+
+/**
+ * Loads view transitions for the current theme, if configured.
+ *
+ * @since 6.9.0
+ */
+function wp_load_view_transitions() {
+	if ( ! current_theme_supports( 'view-transitions' ) ) {
+		return;
+	}
+
+	$stylesheet = file_get_contents( ABSPATH . WPINC . '/view-transitions.css' );
+
+	// Use an inline style to avoid an extra request, also because it's just 1 line of CSS.
+	wp_register_style( 'wp-view-transitions', false, array(), null );
+	wp_add_inline_style( 'wp-view-transitions', $stylesheet );
+	wp_enqueue_style( 'wp-view-transitions' );
+
+	$theme_support = get_theme_support( 'view-transitions' );
+
+	// No point in loading the script if no specific view transition names are configured.
+	if ( ! $theme_support['global-transition-names'] && ! $theme_support['post-transition-names'] ) {
+		return;
+	}
+
+	$config = array(
+		'postSelector'            => $theme_support['post-selector'],
+		'globalTransitionNames'   => $theme_support['global-transition-names'],
+		'postTransitionNames'     => $theme_support['post-transition-names'],
+		'chronologicalSlideInOut' => $theme_support['chronological-slide-in-out'],
+	);
+	$script = file_get_contents( ABSPATH . WPINC . '/view-transitions.js' );
+	$script = str_replace( '{ __PLACEHOLDER__: true }', wp_json_encode( $config, JSON_FORCE_OBJECT ), $script );
```

Comment:
This seems maybe a bit brittle since it depends on this exact string to be present with spaces. I suggest instead that entire JS file be one top-level closure like:

```js
function ( config ) { /*...*/ }
```

Then here the export logic can be something like:

```php
$script = sprintf( "( %s )( %s );", $script, wp_json_encode( $config, JSON_FORCE_OBJECT ) );
```

This will allow you to define the define jsdoc for the `config` param as well, making it easier to maintain the JS file.

Reply:
Makes sense to make this data a function parameter instead of overriding a placeholder.

But I wonder what's the best solution to do that. A top-level closure still has the problem that it only allows for the script to be inlined in the way it's currently done here, but that's not at all common in WP Core. Maybe we can make it so that it can enqueued like a regular script (but would still be inlined by default)? I guess the easiest solution would be to give this function a name and make it available on `window`?

Reply:
> A top-level closure still has the problem that it only allows for the script to be inlined in the way it's currently done here, but that's not at all common in WP Core

What's not common in WP Core? Printing a script inline? It's done for the emoji loader and embeds:

https://github.com/WordPress/wordpress-develop/blob/ac057e6c692e7e88b1758eb8dc6dee2312bb5c7b/src/wp-includes/formatting.php#L5954-L5957

https://github.com/WordPress/wordpress-develop/blob/ac057e6c692e7e88b1758eb8dc6dee2312bb5c7b/src/wp-includes/embed.php#L1093-L1095

The nice thing about having the JS file be just a closure is that by passing the data in like I showed above we avoid polluting the globals.

Reply:
Right, so a named function rather than an anonymous one. So the JS file could be:

```js
function wpViewTransitionsInit( config ) { /*...*/ }
```

Reply:
Updated in https://github.com/WordPress/wordpress-develop/pull/8370/commits/e2e6e5a85fa18368026069d5db3bd42b2312b5ab